src/utils.js

```diff
@@ -1299,5 +1299,49 @@ export function cyrb53Hash(str, seed = 0) {
 export function isAllowZeroCpmBidsEnabled(bidderCode) {
   const bidderSettings = getGlobal().bidderSettings;
   return ((bidderSettings[bidderCode] && bidderSettings[bidderCode].allowZeroCpmBids === true) ||
-   (bidderSettings.standard && bidderSettings.standard.allowZeroCpmBids === true));
+    (bidderSettings.standard && bidderSettings.standard.allowZeroCpmBids === true));
+}
+
+/**
+ * Return a memoized version of the given function that uses the first argument as cache key.
+ */
+export function memoize1(fun) {
+  const cache = {};
+  return function (a0, ...rest) {
+    if (!cache.hasOwnProperty(a0)) {
+      cache[a0] = fun(a0, ...rest);
+    }
+    return cache[a0];
+  }
+}
+
+/**
+ * Return a memoized version of the given zero-arg function.
+ */
+export function cache(fun) {
+  const f = memoize1((arg) => fun());
+  return function () {
+    return f(null);
+  }
+}
+
+/**
+ * Return an object that is the result of recursively merging properties from the given objects. If multiple
+ * objects define the same "leaf" properties, the right-most's value is picked.
+ */
+export function deepMerge(left, right) {
```

Comment:
I just learned that there is already a `mergeDeep` in here. I'll post a revision early next week to slim this down some.